integration-tests/src/setup/test_radio.rs

```diff
@@ -0,0 +1,500 @@
+use crate::utils::{
+    empty_attestation_handler, generate_random_address, get_random_port, setup_mock_env_vars,
+    setup_mock_server, RadioRuntimeConfig,
+};
+use chrono::Utc;
+use colored::*;
+use ethers::signers::LocalWallet;
+use ethers_contract::EthAbiType;
+use ethers_core::types::transaction::eip712::Eip712;
+use ethers_derive_eip712::*;
+use graphcast_sdk::config::{Config, NetworkName, NETWORKS};
+/// Radio specific query function to fetch Proof of Indexing for each allocated subgraph
+use graphcast_sdk::graphcast_agent::GraphcastAgent;
+use graphcast_sdk::graphql::client_graph_node::update_chainhead_blocks;
+use graphcast_sdk::graphql::client_network::query_network_subgraph;
+use graphcast_sdk::graphql::client_registry::query_registry_indexer;
+use graphcast_sdk::{
+    comparison_trigger, graphcast_id_address, read_boot_node_addresses, BlockClock, BlockPointer,
+};
+use hex::encode;
+use num_bigint::BigUint;
+use num_traits::Zero;
+use partial_application::partial;
+use poi_radio::{
+    attestation_handler, compare_attestations, process_messages, save_local_attestation,
+    Attestation, ComparisonResult, LocalAttestationsMap, MessagesArc, RadioPayloadMessage,
+    RemoteAttestationsMap, GRAPHCAST_AGENT, MESSAGES,
+};
+use prost::Message;
+use rand::{thread_rng, Rng};
+use secp256k1::SecretKey;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::env;
+use std::sync::{Arc, Mutex as SyncMutex};
+use std::{thread::sleep, time::Duration};
+use tokio::sync::Mutex as AsyncMutex;
+use tracing::log::warn;
+use tracing::{debug, error, info, trace};
+
+use crate::setup::constants::{MOCK_SUBGRAPH_GOERLI, MOCK_SUBGRAPH_MAINNET};
+use poi_radio::graphql::query_graph_node_poi;
+
+fn round_to_nearest(number: i64) -> i64 {
+    (number / 10) * 10 + if number % 10 > 4 { 10 } else { 0 }
+}
+
+#[derive(Eip712, EthAbiType, Clone, Message, Serialize, Deserialize)]
+#[eip712(
+    name = "Graphcast POI Radio Dummy Msg",
+    version = "0",
+    chain_id = 1,
+    verifying_contract = "0xc944e90c64b2c07662a292be6244bdf05cda44a7"
+)]
+pub struct DummyMsg {
+    #[prost(string, tag = "1")]
+    pub identifier: String,
+    #[prost(int32, tag = "2")]
+    pub dummy_value: i32,
+}
+
+impl DummyMsg {
+    pub fn new(identifier: String, dummy_value: i32) -> Self {
+        DummyMsg {
+            identifier,
+            dummy_value,
+        }
+    }
+}
+
+pub async fn run_test_radio<S, A, P>(
+    runtime_config: &RadioRuntimeConfig,
+    success_handler: S,
+    test_attestation_handler: A,
+    post_comparison_handler: P,
+) where
+    S: Fn(MessagesArc),
+    A: Fn(u64, &RemoteAttestationsMap, &LocalAttestationsMap),
+    P: Fn(MessagesArc, u64, &str, usize),
+{
+    // Parse basic configurations
+    let config = Config::args();
+
+    let collect_message_duration: i64 = env::var("COLLECT_MESSAGE_DURATION")
```

Comment:
yes, but I don't want to run Config::args in `test.radio.rs` since it's already called in `integration-tests/main.rs`. It is easier and makes more sense to use it this way imo because the setup instances set `COLLECT_MESSAGE_DURATION` as an env var too.

Reply:
I think if we make the config editable globally in the future that would help with switching all internal env vars to it, but it's not pressing